TESTING.md

```diff
@@ -0,0 +1,59 @@
+
+### Unit tests
+All unit tests are inside spec folder. You can execute them manually by using the RSpec and specifying the test files, like `rspec spec/path/to/my/tests`.
+
+For pure lib tests the projects use solely RSpec, however, for Chef custom resources tests we use a variation of RSpec, the ChefSpec. Both of them work the same way by issuing the `rspec` command.
+
+#### Coverage
+The unit tests issue a coverage report generated by SimpleCov for the pure lib files, and ChefSpec issues the report for the Chef custom resources (Since SimpleCov does not support it).
+
+However, you must pay attention because SimpleCov's and ChefSpec's coverage report conflict with each other, i.e. as soon ChefSpec's coverage starts running, it will interrupt the SimpleCov's coverage status and it will generate the report at that point. So, because of this, is extremely important running all pure ruby lib files **before** any Chef custom resource, otherwise, the SimpleCov report would be **incorrect**.
+
+The main difference between SimpleCov and ChefSpec coverage is that SimpleCov reports line coverage and ChefSpec shows custom resource touch coverage.
+
+### Style checking
+Style checking is done by `rubocop` for all the custom resource providers and pure ruby lib files, and `foodcritic` to analyze testing recipes and examples.
+
+### Functional tests
+Every resource must have an example for each action that shows how to use it. These examples can be used to do functional and OneView integration testing but they need to be executed manually using the Chef.
+
+## Implementing tests
+All the code must have associated tests, be it the already implemented or new submitted code, and this section covers what tests need to be implemented.
+
+### Unit tests
+The unit tests are required for each pure ruby lib methods and for each Chef custom resource action.
+
+#### Minimum requisites
+The necessary amount of testing is dictated by who is implementing and by who is accepting the code, however, there is a floor acceptance level that must be followed:
+  1. All the pure ruby lib code must have unit tests hitting each line of them, i.e., the line coverage for this kind of code must be 100%.
+  2. Each custom resource must have at least one unit test for each resource action.
+
+#### Lib file tests
+The pure ruby lib files are tested using common RSpec syntax, and they are located in `spec/unit/libraries`.
+
+#### Custom resource tests
+The custom resources are tested using the ChefSpec syntax, similar to RSpec, but they need some extra information to work:
+  1. The actual tests are located in `spec/unit/resource`-like folders. Each resource has a folder, and, inside these folders, there is one test file for each custom resource action (or use case, if necessary).
+  2. The fixtures are fake cookbooks bound to a hardware variant and API version that contain fixture recipes that uniquely executes one action for a specific resource. They are located in `spec/fixtures`.
```

Comment:
The fixtures are fake cookbooks bound to a hardware variant and API version that contain fixture recipes that uniquely execute ~s~ one action for a specific resource. (...)